### DISCUSSION TITLE
intercepting arguments before command execution

### DISCUSSION THREAD

**Discussion by tomvanderputte** [eshaan7/Flask-Shell2HTTP#56] [Q&A] [answered]
Is it possible to intercept and change the dynamic arguments before the command is triggered?
The reason is: the arguments point to input/output files in a certain location. The application that sends the POST requests is not aware of this location (nor do I want it to be), so it sends only the filename. 

So I want to edit the arguments/parameters to predfix the sent arguments with the correct path. How could I achieve this?

**Comment by eshaan7** [comment, marked as answer]
Hi @tomvanderputte,

A way to achieve your use-case is to wrap your command in a script (using sh/bash/python/etc) that modifies the incoming arguments to the desired format before passing them to the actual command. And then expose this script instead of the actual command.
